src/parakeet/scan.py

```diff
@@ -8,13 +8,35 @@
 # This code is distributed under the GPLv3 license, a copy of
 # which is included in the root directory of this package.
 #
-import numpy
+from abc import ABC, abstractmethod
 
+import numpy as np
+from scipy.stats import special_ortho_group
 
-class Scan(object):
+from parakeet.pose import PoseSet
+
+
+class Scan(ABC):
+    """A base class defining the interface for image sampling in a simulation.
     """
-    A class to encapsulate an scan
 
+    @property
+    @abstractmethod
+    def poses(self) -> PoseSet:
+        pass
+
+    @property
+    @abstractmethod
+    def exposure_times(self) -> np.ndarray:
+        pass
+
+    def __len__(self) -> int:
+        """The number of images to sample"""
+        return len(self.poses)
+
+
+class SingleAxisScan(object):
+    """A scan of angles around a single rotation axis.
```

Comment:
Should this inherit from Scan rather than object?

Reply:
Probably! I think I didn't update the `SingleAxisScan` to implement the necessary abstract methods here in favour of waiting for us to discuss coordinate systems